### PR TITLE
V22F-542 Run code coverage over IPCLib

### PR DESCRIPTION
**Pull request by KungFuDonkey**
Adds OpenCppCoverage to ci and generates a code coverage result

### CODE REVIEW

**Comment by KungFuDonkey** [comment]
image

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -45,4 +45,5 @@ jobs:
       with:
         name: test_results_xml
         path: ${{runner.workspace}}/build/IPCTests/test-results/**/*.xml
+
```
Comment (larsdekwant) [inline .github/workflows/ci.yml:48]: random extra line unrelated to this pull-request?
```suggestion
```

**Review by larsdekwant — CHANGES_REQUESTED** [review]
Rename the `code coverage.yml` file to `code-coverage.yml` to avoid any possible errors with spaces in file names.

**Review by suppergerrie2 — APPROVED** [review]

File: IPCTests/Utils.h [hunk 2]
```diff
@@ -15,14 +15,14 @@
...
-/// @brief		Tests if 2 messages are equal to each other
-/// @param msg1 The first message
-/// @param msg2 The second message
-/// @param size The size of the 2 messages
-/// @return		Whether the test failed or succeeded
+/// @brief		  Tests if 2 messages are equal to each other
+/// @param p_msg1 The first message
+/// @param p_msg2 The second message
+/// @param p_size The size of the 2 messages
+/// @return		  Whether the test failed or succeeded
```
Comment (larsdekwant) [inline IPCTests/Utils.h:25]: This is not aligned correctly, probably due to tabs.

File: IPCTests/Utils.h [hunk 3]
```diff
@@ -38,9 +38,9 @@ inline bool TestMessageEqual(const char* p_msg1, const char* p_msg2, int p_size)
     return true;
 }
 
-/// @brief				Generates a random string for testing
-/// @param dataBuffer	The buffer in which the string will be written
-/// @param stringLength The length of the string (must be strictly smaller than the length of the buffer)
+/// @brief				  Generates a random string for testing
+/// @param p_dataBuffer	  The buffer in which the string will be written
+/// @param p_stringLength The length of the string (must be strictly smaller than the length of the buffer)
```
Comment (larsdekwant) [inline IPCTests/Utils.h:43]: Same here, not aligned correctly due to tabs.

**Review by larsdekwant — DISMISSED** [review]

**Pull request merged**